src/System.Management.Automation/utils/MshTraceSource.cs

```diff
@@ -182,13 +184,7 @@ public partial class PSTraceSource
             string description,
             bool traceHeaders)
         {
-            if (string.IsNullOrEmpty(name))
-            {
-                // Note, all callers should have already verified the name before calling this
-                // API, so this exception should never be exposed to an end-user.
-
-                throw new ArgumentException(null, nameof(name));
-            }
+            Dbg.Assert(!string.IsNullOrEmpty(name), "Caller should ensure that name is not null or empty");
```

Comment:
I can not replace runtime checks with debug. Please revert and use `PSArgumentNullException`.

Reply:
The comment on the line above says the caller should verify this parameter and this exception should never be exposed.

Reply:
All our CI tests work on _Release_ only. So we prefer runtime check over Debug.Assert()-s.

Reply:
I think the `Debug.Assert` is arguable but this PR is not the place for this change.